From a17ad748e176bd399b4eef27ec7f904a0488f267 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 29 Nov 2021 11:06:36 +0200
Subject: [PATCH] [SW-0] ppboot: gaudi2: arc: update dccm base

Currently, dccm base is set 0x2706402C (ARC1 DCCM) which is used
for ppboot BSS, stack and data sections. As part of compilation,
data section start is aligned to 8 bytes starting from 0x27064030
instead of 0x2706402C as it. This causes data section to shift by
4 bytes. This patch updates DCCM base to 8 byte aligned address.

Change-Id: I63e53d567f6bb6ea1c96cea104fa3f0f7b7f840f
---

diff --git a/arch/gaudi2/arc/common.h b/arch/gaudi2/arc/common.h
index 7f2c80b..ce8eb9d 100644
--- a/arch/gaudi2/arc/common.h
+++ b/arch/gaudi2/arc/common.h
@@ -67,10 +67,10 @@
 #define BOOT_INFO_BASE          (ARC0_ICCM_BASE + ICCM_BOOT_INFO_OFFSET)
 
 /* In order not to conflict with ARC BOOTROM's usage of DCCM,
- * Let PPBoot start from the offset of 0x402C, for using DCCM
+ * Let PPBoot start from the offset of 0x4030, for using DCCM
  * for its own BSS and Stack.
  */
-#define PPBOOT_DCCM_BASE	0x2706402C
+#define PPBOOT_DCCM_BASE	0x27064030
 
 #define HL_SPI_BASE		0x26000000
 #define HL_CFG_BASE		0xF0000000
